pkgs/games/shipwright/default.nix

```
@@ -0,0 +1,203 @@
+{ stdenv
+, cmake
+, clang
+, lsb-release
+, ninja
+, lib
+, fetchFromGitHub
+, makeDesktopItem
+, copyDesktopItems
+, makeWrapper
+, python3
+, llvmPackages
+, libX11
+, libXrandr
+, libXinerama
+, libXcursor
+, libXi
+, libXext
+, glew
+, SDL2
+, SDL2_net
+, pkg-config
+, libpulseaudio
+, libpng
+, bzip2
+, imagemagick
+, StormLib
+, romVariant ? "debug"
+, mqRomVariant ? "debug-mq"
+, requireFile
+, ootRom ? requireFile {
+    name = "oot-${romVariant}.z64";
+    message = ''
+      This nix expression requires that oot-${romVariant}.z64 is already part of the store.
+      To get this file you can dump your Ocarina of Time's cartridge to a file,
+      and add it to the nix store with nix-store --add-fixed sha256 <FILE>.
+
+      The supported variants are:
+       - debug: Ocarina of Time Debug PAL GC (not Master Quest)
+       - pal-gc: Ocarina of Time PAL GameCube (may lead to crashes and instability)
+
+      This is optional if you have imported an Ocarina of Time Master Quest ROM.
+      If so, please set enableOotRom to false and enableOotMqRom to true.
+      If both are checked, Ship of Harkinian will be built with both ROMs.
+      Alternatively, use the `shipwright-mq` package instead of `shipwright`.
+    '';
+    sha256 = {
+      debug = "94bdeb4ab906db112078a902f4477e9712c4fe803c4efb98c7b97c3f950305ab";
+      pal-gc = "f788793d27aac3f8d91be5f242c4134217c615bfddd5c70384521ea2153435d2";
+    }.${romVariant};
```

Woudl be nice to give a more user-friendly error message if an unsupported variant was set:
```suggestion
    }.${romVariant} or (throw "Unsupported romVariant ${romVariant}. Valid options are 'debug' and 'pal-gc'.");
```